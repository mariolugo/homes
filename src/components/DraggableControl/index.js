import React, { useState } from 'react';
import styled from 'styled-components';

const ControlStyled = styled.div`
  height: 35px;
  box-sizing: border-box;
  width: 290px;
  border-radius: 2px;
  background-color: rgb(255, 255, 255);
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  margin-bottom: 15px;
  cursor: pointer;
  position: absolute;
  bottom: 20px;
  left: 50%;
  margin-left: -148px;
  padding: 8px;
  span {
    font-size: 14px;
  }
`;

const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  // Hide checkbox visually but remain accessible to screen readers.
  // Source: https://polished.js.org/docs/#hidevisually
  border: 0;
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

const Icon = styled.svg`
  fill: none;
  stroke: white;
  stroke-width: 2px;
`;

const StyledCheckbox = styled.div`
  display: inline-block;
  width: 16px;
  height: 16px;
  border: 1px solid black;
  background: ${(props) => (props.checked ? 'rgb(141, 122, 255)' : 'white')};
  border-radius: 3px;
  transition: all 150ms;
  margin-right: 10px;
  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px rgb(141, 122, 255);
  }
  ${Icon} {
    visibility: ${(props) => (props.checked ? 'visible' : 'hidden')};
  }
`;

const CheckboxContainer = styled.div`
  display: inline-block;
  vertical-align: middle;
`;

/**
 * Component used for the map, in here we have a checkbox and we can add logic to handle the checkbox.
 * For example we can activate the checkbox and filter the homes if the map changes
 */
const DraggableControl = () => {
  const [active, setActive] = useState(false);

  const onCheckboxChange = () => {
    setActive((prev) => !prev);
  };

  return (
    <ControlStyled>
      <div>
        <form>
          <label htmlFor="checkbox">
            <CheckboxContainer>
              <HiddenCheckbox id="checkbox" checked={active} onChange={onCheckboxChange} />
              <StyledCheckbox checked={active}>
                <Icon viewBox="0 0 24 24">
                  <polyline points="20 6 9 17 4 12" />
                </Icon>
              </StyledCheckbox>
            </CheckboxContainer>
            <span>Filtrar mientras te mueves en el mapa</span>
          </label>
        </form>
      </div>
    </ControlStyled>
  );
};

export default DraggableControl;
