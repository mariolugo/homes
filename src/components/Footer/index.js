import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { WrapperGrid } from '../';
import FooterLogo from '../../assets/icons/footer-logo.svg';
import Messenger from '../../assets/icons/messenger.svg';

const FooterStyled = styled.div`
  background-color: white;
  border-top: 1px solid ${({ theme }) => theme.colors.grayLight};
  margin: 0px auto;
  min-height: 60px;

  @media only screen and (max-width: 600px) {
    padding-top: 10px;
  }
`;

const LinkStyled = styled.p`
  font-size: 13px;
`;

const FooterLink = styled(LinkStyled)`
  text-align: center;
`;

const ListStyled = styled.ul`
  list-style: none;
  padding-left: 0px;
`;

const ListItemStyled = styled.li`
  svg {
    display: inline-block;
    margin-right: 5px;
    position: relative;
    top: 2px;
  }
  p {
    display: inline-block;
    margin: 0;
  }

  @media only screen and (max-width: 600px) {
    text-align: center;
  }
`;

const LogoContainer = styled(Col)`
  display: flex;
  padding-right: 0;
  align-items: center;
  justify-content: center;
`;

const ColCentered = styled(Col)`
  display: flex;
  align-items: center;
  justify-content: center;
`;

/**
 * Main footer of the application
 */
const Footer = () => (
  <FooterStyled>
    <WrapperGrid>
      <Row>
        <Col xs={12} sm={4}>
          <Row>
            <LogoContainer xs={12} sm={3}>
              <FooterLogo />
            </LogoContainer>
            <Col xs={12} sm={9}>
              <ListStyled>
                <ListItemStyled>
                  <LinkStyled>hey@homie.mx</LinkStyled>
                </ListItemStyled>
                <ListItemStyled>
                  <LinkStyled>5015 1546</LinkStyled>
                </ListItemStyled>
                <ListItemStyled>
                  <Messenger />
                  <LinkStyled>Escríbenos por facebook</LinkStyled>
                </ListItemStyled>
              </ListStyled>
            </Col>
          </Row>
        </Col>
        <Col xs={12} sm={3}>
          <ListStyled>
            <ListItemStyled>
              <LinkStyled>Te buscamos depa</LinkStyled>
            </ListItemStyled>
            <ListItemStyled>
              <LinkStyled>Blog</LinkStyled>
            </ListItemStyled>
            <ListItemStyled>
              <LinkStyled>Equipo</LinkStyled>
            </ListItemStyled>
          </ListStyled>
        </Col>
        <ColCentered xs={12} sm={2}>
          <FooterLink>Aviso de privacidad</FooterLink>
        </ColCentered>
        <ColCentered xs={12} sm={3}>
          <FooterLink>Términos y condiciones</FooterLink>
        </ColCentered>
      </Row>
    </WrapperGrid>
  </FooterStyled>
);

export default Footer;
