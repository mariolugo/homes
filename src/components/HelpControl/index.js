import React from 'react';
import styled from 'styled-components';
import Image from 'next/image';

const ControlStyled = styled.div`
  width: 50px;
  height: 50px;
  background-color: white;
  cursor: pointer;
  position: absolute;
  bottom: 20px;
  right: 50%;
  margin-right: 10px;
  box-sizing: border-box;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  padding: 13px 9px;
  border-radius: 2px;
`;

/**
 * Help control that is required in the map.
 */
const HelpControl = () => {
  return (
    <ControlStyled>
      <Image src="/images/help_map.png" alt="Help Map" width={35} height={35} />
    </ControlStyled>
  );
};

export default HelpControl;
