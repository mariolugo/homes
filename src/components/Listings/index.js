import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { Post } from '../';

const ListingsContainerStyled = styled.div``;

const ListingsContainer = ({ posts = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15] }) => (
  <ListingsContainerStyled>
    <Row>
      {posts.length > 0 &&
        posts.map((p) => (
          <Col key={p} xs={12} md={6}>
            <Post />
          </Col>
        ))}
    </Row>
  </ListingsContainerStyled>
);

export default ListingsContainer;
