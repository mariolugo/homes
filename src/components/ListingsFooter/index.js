import React from 'react';
import styled from 'styled-components';

const ListingsFooterStyled = styled.div`
  p {
    color: ${({ theme }) => theme.colors.textPrimary};
  }
`;

const Title = styled.p`
  text-align: center;
  font-family: 'Open Sans Bold', sans-serif;
`;

const Subtitle = styled.p`
  text-align: center;
`;

const Button = styled.button`
  border-radius: 2px;
  border: none;
  width: 300px;
  display: block;
  height: 35px;
  background-color: ${({ theme }) => theme.colors.secondary};
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  color: white;
  font-family: 'Open Sans Bold', sans-serif;
  line-height: 19px;
  padding: 8px 0px;
  margin: 20px auto;
`;

const ListingsFooter = () => (
  <ListingsFooterStyled>
    <Title>¿No encuentras inmueble?</Title>
    <Subtitle>Dinos qué quieres y nosotros te lo buscamos</Subtitle>
    <Button>Encuéntrenme un inmueble</Button>
  </ListingsFooterStyled>
);

export default ListingsFooter;
