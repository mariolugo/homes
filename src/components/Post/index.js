import React, { useRef } from 'react';
import styled from 'styled-components';
// import { Col, Row } from 'react-styled-flexboxgrid';
import FavoriteOutline from '../../assets/icons/favorite-outline.svg';
import Slider from 'react-styled-carousel';
import Image from 'next/image';
import { useGetWidth, formatPrice } from '../../utils';
import { homeTypes } from '../../types';
import PropTypes from 'prop-types';
const PostContainer = styled.div`
  background-color: white;
  margin-bottom: 25px;
  border-radius: 2px;
  transition: all 0.2s ease-in-out 0s;
  position: relative;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 2px 4px 0px;
  cursor: pointer;
  &:hover {
    box-shadow: rgba(0, 0, 0, 0.5) 0px 4px 8px 0px;
  }
`;

const PostPrice = styled.div`
  width: 111px;
  height: 35px;
  display: block;
  position: absolute;
  left: 0px;
  top: 0px;
  padding: 6px 7px;
  box-sizing: border-box;
  font-family: 'Open Sans Bold', sans-serif;
  z-index: 10;
  background-color: white;
`;

const FavoriteIcon = styled.button`
  cursor: pointer;
  border: none;
  background-color: transparent;
  float: right;
  outline: none;
  margin: 0;
  svg {
    width: 1.2em;
    position: relative;
    top: 2px;
  }
`;

const Price = styled.h3`
  font-size: 16px;
  float: left;
  margin: 0;
  color: ${({ theme }) => theme.colors.secondary};
`;

const ImageSlide = styled.div`
  width: 100%;
  height: 150px;
  position: relative;

  img {
    width: 100%;
  }
  @media only screen and (max-width: 500px) {
    height: 210px;
  }
`;

const ExclusiveContainer = styled.div`
  width: 100%;
  height: 22px;
  display: block;
  position: absolute;
  left: 0px;
  bottom: 0px;
  padding: 0px;
  box-sizing: border-box;
  font-family: 'Open Sans Bold', sans-serif;
  z-index: 10;
  background-color: ${({ theme }) => theme.colors.secondaryTransparent};
  color: white;

  p {
    margin: 0;
    text-align: center;
    font-size: 14px;
  }
`;

const PostTitle = styled.p`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 8px;
  font-size: 14px;
  color: ${({ theme }) => theme.colors.textPrimary};
`;

const Amenities = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const Amenity = styled.li`
  display: inline-block;
  width: ${(props) => props.width / 5}px;
  height: 32px;
  font-size: 14px;
  padding: 8px 3px;
  box-sizing: border-box;
  border-right: 1px solid ${({ theme }) => theme.colors.grayLight};
`;

const AmenityTitle = styled.p`
  display: inline-block;
  margin: 0;
  margin-right: 5px;
  margin-left: 3px;
  font-size: 13px;
  position: relative;
  bottom: 2px;
`;

const AmenityImg = styled(Image)`
  vertical-align: middle;
  display: inline-block;
  img {
    position: relative;
    top: 5px;
  }
`;

const LeftArrow = styled.button`
  color: #fff;
  position: absolute;
  margin-left: 3px;
  margin-top: 0px;
  width: 16px;
  background-color: #fff;
  left: 0px;

  border: 0;
  cursor: pointer;
  outline: none;
  z-index: 1;
  height: 100%;
  background-color: transparent;
  :before {
    content: '';
    position: absolute;
    left: 1px;
    top: calc(50% - 5px);
    width: 10px;
    height: 10px;
    border-top: solid 2px #fff;
    border-right: solid 2px #fff;
    transform: rotate(-135deg);
  }
  :after {
    content: '';
    position: absolute;
  }

  &:hover {
    background-color: ${({ theme }) => theme.colors.textPrimary};
    opacity: 0.39;
  }
`;

const RightArrow = styled.button`
  color: #fff;
  position: absolute;
  margin-right: 3px;
  margin-top: 0px;
  width: 16px;
  background-color: #fff;
  right: 0px;

  border: 0;
  cursor: pointer;
  outline: none;
  z-index: 1;
  height: 100%;
  background-color: transparent;
  :before {
    content: '';
    position: absolute;
    right: 1px;
    top: calc(50% - 5px);
    width: 10px;
    height: 10px;
    border-top: solid 2px #fff;
    border-right: solid 2px #fff;
    transform: rotate(45deg);
  }
  :after {
    content: '';
    position: absolute;
  }

  &:hover {
    background-color: ${({ theme }) => theme.colors.textPrimary};
    opacity: 0.39;
  }
`;

const amenities = ['bedrooms', 'bathrooms', 'parkings', 'pet_friendly', 'sqare_mts'];

const responsive = [
  { breakPoint: 1280, cardsToShow: 1 }, // this will be applied if screen size is greater than 1280px. cardsToShow will become 4.
  { breakPoint: 760, cardsToShow: 1 },
];

const renderAmenity = (type, value) => {
  switch (type) {
    case 'bedrooms':
      return (
        <div>
          <AmenityTitle>{value[type]}</AmenityTitle>
          <AmenityImg src="/images/bedroom.png" width={18} height={15} alt="bedroom" />
        </div>
      );
    case 'bathrooms':
      return (
        <div>
          <AmenityTitle>{value[type]}</AmenityTitle>
          <AmenityImg src="/images/bathrooms.png" width={18} height={15} alt="bathrooms" />
        </div>
      );
    case 'parkings':
      return (
        <div>
          <AmenityTitle>{value[type]}</AmenityTitle>
          <AmenityImg src="/images/parking.png" width={18} height={15} alt="parking" />
        </div>
      );
    case 'pet_friendly':
      return (
        <div>
          <AmenityTitle>{value[type] ? 'Si' : 'No'}</AmenityTitle>
          <AmenityImg src="/images/pet.png" width={18} height={15} alt="pet friendly" />
        </div>
      );
    case 'sqare_mts':
      return (
        <div>
          <AmenityTitle>
            {value[type]}m<sup>2</sup>
          </AmenityTitle>
        </div>
      );
    default:
      return null;
  }
};

const Post = ({
  abbr_address,
  bathrooms,
  bedrooms,
  is_homie_exclusive,
  name,
  parkings,
  pet_friendly,
  photos,
  price,
  sqare_mts,
  id,
  onHover,
}) => {
  const componentRef = useRef();
  const width = useGetWidth(componentRef);

  return (
    <PostContainer ref={componentRef} onMouseEnter={() => onHover(id)}>
      <PostPrice>
        <Price>${formatPrice(price)}</Price>
        <FavoriteIcon>
          <FavoriteOutline />
        </FavoriteIcon>
      </PostPrice>
      <Slider
        showDots={false}
        padding={'0'}
        responsive={responsive}
        LeftArrow={<LeftArrow />}
        RightArrow={<RightArrow />}
        autoSlide={false}
        DotsWrapper={() => null}
        cardsToShow={1}>
        {photos.map((p) => (
          <ImageSlide key={p}>
            <img src={p} alt={name} />
            {is_homie_exclusive && (
              <ExclusiveContainer>
                <p>Exclusivo de Homie</p>
              </ExclusiveContainer>
            )}
          </ImageSlide>
        ))}
      </Slider>

      <div>
        <PostTitle>{abbr_address}</PostTitle>
      </div>
      <Amenities>
        {amenities.map((amenity) => (
          <Amenity key={amenity} width={width}>
            {renderAmenity(amenity, {
              bathrooms,
              pet_friendly,
              bedrooms,
              parkings,
              sqare_mts,
            })}
          </Amenity>
        ))}
      </Amenities>
    </PostContainer>
  );
};

Post.propTypes = {
  ...homeTypes,
  onHover: PropTypes.func,
};

export default Post;
