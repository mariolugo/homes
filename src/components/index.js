import NavBar from './NavBar';
import Listings from './Listings';
import Map from './Map';
import Footer from './Footer';
import Post from './Post';
import WrapperGrid from './WrapperGrid';

export { NavBar, Listings, Map, Footer, Post, WrapperGrid };
