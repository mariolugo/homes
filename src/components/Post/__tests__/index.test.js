import { mountWithTheme } from '../../../../tests/themeProvider';
import { Post } from '../../../components';

/** @test {Post Component} */
describe('Post Component', () => {
  it('should render without crashing', () => {
    const wrapper = mountWithTheme(<Post />);
    expect(wrapper).toMatchSnapshot();
  });
});
