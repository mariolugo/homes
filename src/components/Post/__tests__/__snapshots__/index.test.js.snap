// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-kfzAmx ccMhrA"
      >
        <div
          class="sc-dIUggk kFoRgP"
        >
          <h3
            class="sc-dmlrTW dHnphj"
          >
            $
            NaN
          </h3>
          <button
            class="sc-hHftDr NjyMJ"
          >
            <svg
              class="svg-inline--fa fa-heart fa-w-16"
              color="#ff1555"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div>
          <div
            class="sc-ifAKCX kdmnHy"
          >
            <div
              class="sc-bdVaJa TqLrp"
            >
              <div
                class="sc-EHOje jAMlvv"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="sc-iwyYcG kqCFOt"
          />
        </div>
        <ul
          class="sc-cxFLnm cyMBUO"
        >
          <li
            class="sc-lmoMRL lgTTIG"
            width="0"
          >
            <div>
              <p
                class="sc-iJuUWI eyKIYN"
              />
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="bedroom"
                  class="sc-giIncl eduGjq"
                  decoding="async"
                  src="/_next/image?url=%2Fimages%2Fbedroom.png&w=320&q=75"
                  srcset="/_next/image?url=%2Fimages%2Fbedroom.png&w=320&q=75 1x"
                  style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </li>
          <li
            class="sc-lmoMRL lgTTIG"
            width="0"
          >
            <div>
              <p
                class="sc-iJuUWI eyKIYN"
              />
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="bathrooms"
                  class="sc-giIncl eduGjq"
                  decoding="async"
                  src="/_next/image?url=%2Fimages%2Fbathrooms.png&w=320&q=75"
                  srcset="/_next/image?url=%2Fimages%2Fbathrooms.png&w=320&q=75 1x"
                  style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </li>
          <li
            class="sc-lmoMRL lgTTIG"
            width="0"
          >
            <div>
              <p
                class="sc-iJuUWI eyKIYN"
              />
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="parking"
                  class="sc-giIncl eduGjq"
                  decoding="async"
                  src="/_next/image?url=%2Fimages%2Fparking.png&w=320&q=75"
                  srcset="/_next/image?url=%2Fimages%2Fparking.png&w=320&q=75 1x"
                  style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </li>
          <li
            class="sc-lmoMRL lgTTIG"
            width="0"
          >
            <div>
              <p
                class="sc-iJuUWI eyKIYN"
              >
                No
              </p>
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="pet friendly"
                  class="sc-giIncl eduGjq"
                  decoding="async"
                  src="/_next/image?url=%2Fimages%2Fpet.png&w=320&q=75"
                  srcset="/_next/image?url=%2Fimages%2Fpet.png&w=320&q=75 1x"
                  style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </li>
          <li
            class="sc-lmoMRL lgTTIG"
            width="0"
          >
            <div>
              <p
                class="sc-iJuUWI eyKIYN"
              >
                m
                <sup>
                  2
                </sup>
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-kfzAmx ccMhrA"
    >
      <div
        class="sc-dIUggk kFoRgP"
      >
        <h3
          class="sc-dmlrTW dHnphj"
        >
          $
          NaN
        </h3>
        <button
          class="sc-hHftDr NjyMJ"
        >
          <svg
            class="svg-inline--fa fa-heart fa-w-16"
            color="#ff1555"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div>
        <div
          class="sc-ifAKCX kdmnHy"
        >
          <div
            class="sc-bdVaJa TqLrp"
          >
            <div
              class="sc-EHOje jAMlvv"
            />
          </div>
        </div>
      </div>
      <div>
        <p
          class="sc-iwyYcG kqCFOt"
        />
      </div>
      <ul
        class="sc-cxFLnm cyMBUO"
      >
        <li
          class="sc-lmoMRL lgTTIG"
          width="0"
        >
          <div>
            <p
              class="sc-iJuUWI eyKIYN"
            />
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                  style="max-width: 100%; display: block;"
                />
              </div>
              <img
                alt="bedroom"
                class="sc-giIncl eduGjq"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Fbedroom.png&w=320&q=75"
                srcset="/_next/image?url=%2Fimages%2Fbedroom.png&w=320&q=75 1x"
                style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </li>
        <li
          class="sc-lmoMRL lgTTIG"
          width="0"
        >
          <div>
            <p
              class="sc-iJuUWI eyKIYN"
            />
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                  style="max-width: 100%; display: block;"
                />
              </div>
              <img
                alt="bathrooms"
                class="sc-giIncl eduGjq"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Fbathrooms.png&w=320&q=75"
                srcset="/_next/image?url=%2Fimages%2Fbathrooms.png&w=320&q=75 1x"
                style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </li>
        <li
          class="sc-lmoMRL lgTTIG"
          width="0"
        >
          <div>
            <p
              class="sc-iJuUWI eyKIYN"
            />
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                  style="max-width: 100%; display: block;"
                />
              </div>
              <img
                alt="parking"
                class="sc-giIncl eduGjq"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Fparking.png&w=320&q=75"
                srcset="/_next/image?url=%2Fimages%2Fparking.png&w=320&q=75 1x"
                style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </li>
        <li
          class="sc-lmoMRL lgTTIG"
          width="0"
        >
          <div>
            <p
              class="sc-iJuUWI eyKIYN"
            >
              No
            </p>
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,<svg width=\\"18\\" height=\\"15\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                  style="max-width: 100%; display: block;"
                />
              </div>
              <img
                alt="pet friendly"
                class="sc-giIncl eduGjq"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Fpet.png&w=320&q=75"
                srcset="/_next/image?url=%2Fimages%2Fpet.png&w=320&q=75 1x"
                style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </li>
        <li
          class="sc-lmoMRL lgTTIG"
          width="0"
        >
          <div>
            <p
              class="sc-iJuUWI eyKIYN"
            >
              m
              <sup>
                2
              </sup>
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
